tests/test_transactions_reader.py

```diff
@@ -41,19 +42,21 @@ def test_transactions_reader_columns_error(testfiles_dir):
     transactions = transactions_to_list([transaction])
     # Add an extra column to the transaction
     transactions[0].append('EXTRA_COLUMN_VALUE')
-    with pytest.raises(ClickException):
-        filepath = create_csv_file(testfiles_dir,
-                                   "too_many_cols.csv",
-                                   transactions,
-                                   True)
+    filepath = create_csv_file(testfiles_dir,
+                               "too_many_cols.csv",
+                               transactions,
+                               True)
+    with pytest.raises(ClickException) as excinfo:
         TransactionsReader.get_transactions(filepath)
+    assert excinfo.value.message == "Transaction entry 0: expected 8 columns, entry has 9"  # noqa: E501
 
 
 def test_transactions_reader_file_not_found_error(testfiles_dir):
     """Testing TransactionsReader with a non-existent file"""
-    with pytest.raises(ClickException):
+    with pytest.raises(ClickException) as excinfo:
         TransactionsReader.get_transactions(create_csv_file(testfiles_dir,
                                                             "dne.csv"))
+    assert "File not found" in excinfo.value.message
```

Comment:
You may as well just include "testfiles_dir" in the error message expectation string and assert that it equals `excinfo.value.message`. There's nothing wrong with doing that.

Reply:
Fixed